### DISCUSSION TITLE
Running mathesar alongside other services: using a pre-existing Caddy?

### DISCUSSION THREAD

**Discussion by amca01** [mathesar-foundation/mathesar#3557] [User Q&A] [answered]
In the hope of using mathesar with my other services, I've switched from SWAG to Caddy for my reverse proxy, and I've now got everything working (except for one service).   However, the mathesar docker compose file comes preloaded, as it were, with its own Caddy container.  If I already have Caddy up and running, is there a more "bare-bones" docker compose I can use, and an appropriate Caddyfile block?
Many thanks,
Alasdair

**Comment by Anish9901** [comment, marked as answer]
Thank you for reaching out about this @amca01, I hope the discussion we had at https://github.com/mathesar-foundation/mathesar/issues/3527#issuecomment-2082454331 helped you get your desired setup for mathesar. Please let us know if you have any more questions.

**Comment by amca01** [comment]
Yes, indeed, and with many thanks @Anish9901 .  Your help has been invaluable, and as I said on the other discussion, I'm thrilled to finally have mathesar up and running with my other services.  

As a separate issue, and maybe to help usability, it would be good to have mathesar work with other reverse proxies: nginx/SWAG, traefik, and so on.  I was using SWAG very happily, but to use mathesar I needed to switch everything to Caddy, which I did.  A nice thing about SWAG is that configuration files for many services are already available.  It's more verbose than Caddy,  but in fact I had less trouble setting it up than I did with Caddy.  (And there is still one service I can't get to work, even with lots of help and advice.)

**Comment by Anish9901** [comment]
Thanks for the feedback! I do understand that switching the entire config of your infrastructure for a service could be annoying. That said, Enhancing Mathesar's compatibility with other reverse proxies like nginx/SWAG and Traefik is definitely on our radar and we are constantly working towards supporting more versatile setups.

**Reply by kgodey** [reply]
@Anish9901 can you create an issue to track the work of figuring out Mathesar configuration with those reverse proxies?

**Reply by Anish9901** [reply]
Tracking in #3601
